x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,2862 @@
+    - **Remove element**. (A _ B) and (A _ A)
+  - If element index exists in current and target version, but not in base:
+    - If base equals current: **add element/keep target**. (_ A A)
+    - If base differs from current: **keep both current and target**. (_ A B)
+  - If element index exists only in base version:
+    - **Remove element** (A _ _)
+  - If element index exists only in current version:
+    - **Keep current version** (_ A _)
+  - If element index exists only in target version:
+    - **Keep target version** (_ _ A)
+    
+<table>
+    <thead>
+      <tr>
+        <th style="border-right:3px solid black">Use case</th>
+        <th>Base version</th>
+        <th>Current version</th>
+        <th style="border-right:3px solid black">Target version</th>
+        <th>Merged version (output)</th>
+      </tr>
+    </thead>
+  <tbody valign="top">
+    <tr>
+      <td rowspan=2 style="border-right:3px solid black">Keep user changes if element conflicts, but apply changes in elements with clean updates <br><br> <ul><li><b>1st elem: (A B C - keep current) </li><li>2nd elem: (A A B - keep target)</b></li></ul><br><br> Same for: <br><ul><li><b>1st elem: (A B C) + 2nd elem: (A B A - keep current)</b></li><li><b>1st elem: (A B C) + 2nd elem: (A B B - keep current/target)</b></li></ul></td>
+      <td>
+      <pre>
+[
+  logs-*,
+  logstash-*
+]
+        </pre>
+      </td>
+      <td>
+        <pre>
+[
+  cluster_one:logs-*,
+  cluster_one:logstash-*
+]
+        </pre>
+      </td>
+      <td style="border-right:3px solid black">
+      <pre>
+[
+  new-index1-*,
+  new-index2-*,
+]
+        </pre>
+        </td>
+      <td>
+      <pre>
+[
+  cluster_one:logs-*,
+  cluster_one:logstash-*
+]
+        </pre>
+      </td>
+    </tr>
+    <tr>
+      <td>
+      <pre>
+[
+  {
+    ecs: true,
+    name: "event.action",
+    type: "unknown"
+  },
+  {
+    ecs: false,
+    name: "host.os.type",
+    type: "keyword"
+  },
+]
+        </pre>
+      </td>
+      <td>
+        <pre>
+[
+  {
+    ecs: true,
+    name: "event.UPDATED",
+    type: "unknown"
+  },
+  {
+    ecs: false,
+    name: "host.os.type",
+    type: "keyword"
+  },
+]
+        </pre>
+      </td>
+      <td style="border-right:3px solid black">
+      <pre>
+[
+  {
+    ecs: true,
+    name: "event.action",
+    type: "keyword"
+  },
+  {
+    ecs: false,
+    name: "host.user.id",
+    type: "keyword"
+  },
+]
+        </pre>
+        </td>
+      <td>
+      <pre>
+[
+  {
+    ecs: true,
+    name: "event.UPDATED",
+    type: "unknown"
+  },
+  {
+    ecs: false,
+    name: "host.user.id",
+    type: "keyword"
+  },
+]
+        </pre>
+      </td>
+    </tr>
+    <tr>
+      <td rowspan=2 style="border-right:3px solid black">Keep user changes if element conflicts, and remove any <b>uncustomized</b> elements if removed in update <br><br><ul><li><b>1st elem: (A B C - keep current) </li><li>2nd elem: (A A _ - remove element)</b></li></ul></b></td>
+      <td>
```

Comment:
I think this is an overcomplication. If there's any conflict between current and target changes, we should mark the result as a conflict and simply set `merged = current` without trying to do a clever half-merge. Because:

- It would be likely incorrect anyway.
- It would require us to implement and maintain some clever code.

I'd like to go through the examples in this table together over zoom to give a quick idea how I think they all should be processed by a diff algorithm.

Reply:
I agree we could first implement a simpler predictable merged field logic, and then improve it in the following iteration.